src/components/settings-security/settings-security.tsx

```diff
@@ -0,0 +1,113 @@
+import React, {useCallback, useRef, useState} from 'react';
+
+import {Alert, StyleSheet, Switch, View} from 'react-native';
+
+import {Color} from '@app/colors';
+import {Pin, PinInterface} from '@app/components/pin';
+import {MenuNavigationButton, Spacer, Text} from '@app/components/ui';
+import {useApp} from '@app/hooks';
+import {getText, I18N} from '@app/i18n';
+import {BiometryType} from '@app/types';
+
+const biometryName = {
+  [BiometryType.faceId]: 'Face ID',
+  [BiometryType.touchId]: 'Touch ID',
+  [BiometryType.fingerprint]: 'Fingerprint',
+  [BiometryType.unknown]: 'unknown',
+};
+
+interface SettingsSecurityProps {
+  onSubmit?: () => void;
+}
+
+export const SettingsSecurity = ({
+  onSubmit = () => {},
+}: SettingsSecurityProps) => {
+  const app = useApp();
+  const [biometry, setBiometry] = useState(app.biometry);
+  const [loggedIn, setLoggedIn] = useState(false);
+  const pinRef = useRef<PinInterface>();
+
+  const onPin = useCallback(
+    async (pin: string) => {
+      try {
+        await app.comparePin(pin);
+        setLoggedIn(true);
+      } catch (e) {
+        pinRef.current?.reset('wrong pin');
+      }
+    },
+    [app],
+  );
+
+  const onToggleBiometry = useCallback(async () => {
+    if (!biometry) {
+      try {
+        await app.biometryAuth();
+        app.biometry = true;
+        setBiometry(true);
+      } catch (e) {
+        if (e instanceof Error) {
+          Alert.alert(e.message);
+        }
+        app.biometry = false;
+        setBiometry(false);
+      }
+    } else {
+      app.biometry = false;
+      setBiometry(false);
+    }
+  }, [app, biometry]);
+
+  if (!loggedIn) {
+    return <Pin title={getText(I18N.settingsSecurityWalletPin)} onPin={onPin} ref={pinRef} />;
```

Comment:
It would be nice if Pin accepted `i18nTitle`
```jsx
<Pin i18nTitle={I18N.settingsSecurityWalletPin} onPin={onPin} ref={pinRef} />
```

Reply:
Done!
- add `i18n `and `i18params `for `<Pin/>`